climpred/classes.py

```diff
@@ -230,18 +400,32 @@ def generate_uninitialized(self, var=None):
         """
         if var is not None:
             uninit = bootstrap_uninit_pm_ensemble_from_control(
-                self.initialized[var], self.control[var]
+                self._datasets['initialized'][var], self._datasets['control'][var]
             ).to_dataset()
         else:
             uninit = bootstrap_uninit_pm_ensemble_from_control(
-                self.initialized, self.control
+                self._datasets['initialized'], self._datasets['control']
             )
-        self.uninitialized = uninit
-
-    def compute_metric(self, metric='pearson_r', comparison='m2m'):
+        datasets = self._datasets.copy()
+        datasets.update({'uninitialized': uninit})
+        return self._construct_direct(datasets)
+
+    # -----------------
+    # Getters & Setters
+    # -----------------
+    def get_control(self):
+        """Returns the control as an xarray dataset."""
+        return self._datasets['control']
+
+    # ------------------
+    # Analysis Functions
+    # ------------------
+    def compute_metric(self, var=None, metric='pearson_r', comparison='m2m'):
```

Comment:
Was there no way in compute_skill without specifying the var argument? Before we could compute skill over the whole dataset, ie. sst and sss if both available.

Reply:
If you don't use the `var` argument it just computes it over the whole dataset. I guess I was thinking this would be nice to have if you have a lot of variables and don't want to take the computational cost to do all of them.

But right now it drops all variables that the datasets don't have in common and computes over the intersection of that set.

**Should I just drop the var keyword all together?** I figured it could be something like `compute_metric(var=['SST', 'SSH'])`, `compute_metric(var='SST'`) or just no argument there.

Reply:
would it work that the user first selects the dataarray by calling `HindEnsemble['SST].compute_metric()`? then I would drop it. otherwise not.

Reply:
I think that's a better way to do it and would require something with `__getattr__`. Right now it recognizes `.func()` , e.g. `.isel()`. I might bump this to a later PR.

Reply:
Currently you cannot subscript by `HindcastEnsemble['SST']` etc. I'll see if it's an easy fix but might be for a later PR.

Reply:
But I'll drop the var= argument for now.